Scripts/Global/App.gd

```diff
@@ -0,0 +1,24 @@
+extends Node
+
+
+# The DPI of a 23.8 inch display with a 1920x1080 resolution
+const BASE_SCALE_DPI = 92.56
+
+
+func _ready() -> void:
+	update_window(true)
+	get_window().connect("size_changed", update_window)
+
+
+func update_window(update_size: bool = false):
+	var screen_dpi = DisplayServer.screen_get_dpi()
+	var screen_size = DisplayServer.window_get_size()
+	# Guess the screen scale because [DisplayServer.screen_get_scale()] only work on macOS and Linux.
+	var guessed_screen_scale: float = min(round(screen_dpi/BASE_SCALE_DPI), 2.0)
+	
+	ProjectSettings.set_setting("display/window/stretch/scale", guessed_screen_scale)
+	ProjectSettings.save_custom("user://override.cfg")
```

Comment:
This doesn't do anything when you export the project, it only works when you run Monologue from the Godot editor. When using `override.cfg`, it should **"be used in exported projects by placing this file in the same directory as the project binary"**, not in the user directory, see [ProjectSetttings overriding](https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#projectsettings).

> To configure the stretch scale at runtime from a script, use the `get_tree().root.content_scale_factor` property (see [Window.content_scale_factor](https://docs.godotengine.org/en/stable/classes/class_window.html#class-window-property-content-scale-factor)).